js/usercss.js

```diff
@@ -0,0 +1,183 @@
+/* globals loadScript mozParser */
+
+'use strict';
+
+function wildcard2regexp(text) {
+  return text.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&').replace(/\\\*/g, '.*');
+}
+
+
+function guessType(value) {
+  if (/^url\(.+\)$/i.test(value)) {
+    return 'image';
+  }
+  if (/^#[0-9a-f]{3,8}$/i.test(value)) {
+    return 'color';
+  }
+  if (/^hsla?\(.+\)$/i.test(value)) {
+    return 'color';
+  }
+  if (/^rgba?\(.+\)$/i.test(value)) {
+    return 'color';
+  }
+  // should we use a color-name table to guess type?
+  return 'text';
+}
+
+const BUILDER = {
```

Comment:
we don't need a global variable with an ambiguous name, move it into usercss object for example